pkg/utils/file/compress_test.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+	"testing"
+)
+
+func TestCompress(t *testing.T) {
+	tests := []struct {
+		name     string
+		ext      string
```

Comment:
```suggestion
		ext      string
		data     []byte
```

Don't hard-coded the data

Reply:
Shall I need to have separate data for all test cases?